landoapi/hgexports.py

```diff
@@ -605,3 +616,144 @@ def run_diff_checks(self) -> list[str]:
                 issues.append(issue)
 
         return issues
+
+
+@dataclass
+class PushCheck:
+    """Provides an interface to implement push-wide checks."""
+
+    repo: Repo
+
+    def relevant(self) -> bool:
+        """Return `True` if the check is relevant for the given repo."""
+        raise NotImplementedError()
+
+    def next_diff(self, patch_helper: PatchHelper):
+        """Pass the next diff to be checked into the check and run."""
+        raise NotImplementedError()
+
+    def result(self) -> Optional[str]:
+        """Return the result of the check."""
+        raise NotImplementedError()
+
+
+@dataclass
+class BugReferencesCheck(PushCheck):
+    """Prevent commit messages referencing non-public bugs from try."""
+
+    bug_ids: set[int] = field(default_factory=set)
+    skip_check: bool = False
+
+    def relevant(self) -> bool:
+        if self.repo.tree == "try":
+            return True
+
+        return False
+
+    def next_diff(self, patch_helper: PatchHelper):
+        commit_message = patch_helper.get_commit_description()
```

Comment:
Missing doc string (parent class doesn't really explain what is happening here either as nothing is being "run").

I'm not sure this method is doing what it's intended to do? Or maybe it just has the wrong name. What is being done in this method may be better done when initializing this check instance.